### ISSUE TITLE
stdin/stdout/stderr constants not defined

### ISSUE THREAD

**Issue by dktapps** [krakjoe/pthreads#806]
Environment
==========

  * PHP: 7.2.1
  * pthreads: 3.1.7dev
  * OS: windows

Summary
======
This issue is pretty much the same complaint as seen in https://github.com/krakjoe/pthreads/issues/44 . I'm making this issue because this is a problem when wanting to use library code that needs these constants and subsequently crashes because they aren't defined.

These can be defined manually in userland to work around this problem, but that is extra boilerplate for every thread class created. Would it not be more sensible to have a solution for this in pthreads itself?

I see that @krakjoe mentions that not all threads created in userland will need these constants, but that is also true of lots of thread-unaware library code. For code that _does_ need it it's a headache when these are not present. I was unaware that these constants were not copied/recreated before looking at pthreads' source code.

Reproducing Code
==============
```php
<?php

$thread = new class extends \Thread{
    public function run(){
        var_dump(defined('STDOUT'));
    }
}

$thread->start(PTHREADS_INHERIT_NONE) && $thread->join();
```

Expected Output
=============
```
bool(true)
```

Actual Output
============
```
bool(false)
```

**Comment by SirSnyder** [comment]
Referring to #44, since pthreads doesn't support any web server(http) sapi anymore, shouldn't it be ok to expose stdin/stdout/stderr to threads?

**Comment by dktapps** [comment]
I would assume so. For example my application has a (blocking) threaded console reader which reads user input typed into the console.

Since these constants are not defined, I just workaround the issue by opening resource handles.  A simple userland workaround for this is to just:
```php
define('STDIN', fopen('php://stdin', 'rb'));
define('STDOUT', fopen('php://stdout', 'wb'));
define('STDERR', fopen('php://stderr', 'wb'));
```

[These](https://github.com/krakjoe/pthreads/blob/master/php_pthreads.c#L99) are the currently supported SAPIs.

**Comment by dktapps** [comment]
According to http://php.net/manual/en/features.commandline.io-streams.php, only the CLI SAPI (and phpdbg, but that isn't mentioned) defines these constants, so I don't see a problem anyway. They can simply be recreated on the child thread if they were found to exist on the parent thread, no?

**Comment by SirSnyder** [comment]
Merged as 8f4205d
